### PR TITLE
Fix 14355

### PR DESCRIPTION
**Pull request by neharathod**

### CODE REVIEW

**Comment by williambr** [comment]
As you can see the CI has detected new PEP8 failures in your commit which you should address before we merge this.

Also, being a bit pedantic, you have two commits with same log, which is not very helpful.

**Comment by neharathod** [comment]
Hi, I am using the PEP8 tool (https://pypi.python.org/pypi/pep8) for PEP8 compliance and I see no such PEP8 failures. Please can I have the CI logs generated at your end to understand the failures in my commit.

**Comment by williambr** [comment]
The log is in this very ticket: https://travis-ci.org/freenas/freenas/builds/126030068 ... Just see below...

` gui/middleware/zfs.py:46:9: W503 line break before binary operator
 gui/plugins/views.py:466:17: W503 line break before binary operator
 gui/storage/admin.py:135:1: W293 blank line contains whitespace
+gui/storage/views.py:415:61: E231 missing whitespace after ','
+gui/storage/views.py:415:92: E231 missing whitespace after ','
+gui/storage/views.py:523:58: E231 missing whitespace after ','
+gui/storage/views.py:523:89: E231 missing whitespace after ','
 gui/storage/__init__.py:1:1: E265 block comment should start with '# '
 gui/storage/__init__.py:27:1: W391 blank line at end of file
 gui/storage/models.py:232:21: W503 line break before binary operator`

**Comment by skarekrow** [comment]
You will note that it says "Some checks have failed" below your comments.  Click the details to the right of that and then scroll down and click view log. You will see the output @williambr has helpfully pasted.

**Comment by neharathod** [comment]
As I see the logs, I understand that the 4 PEP8 failures for gui/storage/views.py are related to my commit, whereas I have not made any changes to the rest of the files mentioned.

**Comment by williambr** [comment]
The only lines it cares about are the ones who changed (e.g. the ones that starts with the sign "+").
That means new errors introduced, by your commit, yes.

**Comment by neharathod** [comment]
Also, is it advisable to create a new branch, as I already have 2 commits made on the current one.

**Comment by williambr** [comment]
No, just add a new commit to current branch, but be verbose about what you did in the commit, dont repeat the text as you did in the early ones.

**Comment by neharathod** [comment]
Okay, Thank you!

**Comment by williambr** [comment]
Not quire right yet.

Setting user property is not working. E.g. If I have a pool tank and try to edit the dataset tank, comment will not stick (no org.freenas:description zfs property is set)

**Comment by williambr** [comment]
Also, when creating a dataset comments field is required, it should not be.

**Comment by neharathod** [comment]
I have now added the code to set the comments property in the Edit forms of dataset/zvols.
I have also made the 'comments' field optional.

**Comment by williambr** [comment]
One more issue.

If you set a comment of a top level dataset .e.g tank as "test", it will show it as inherited for child datasets if you set nothing. For example, tank/jails will show "inherit (ok)".

This is not expected behavior for the UI, if the source of the org.freenas:description property is "inherited" it should not show anything in the UI.

**Comment by neharathod** [comment]
Yes, but this happens automatically when a jail is created for a particular dataset.

**Comment by williambr** [comment]
Yes, it will happen, and I saying it should not show in the datagrid for inherited comment property.

File: gui/api/resources.py [hunk 1]
```diff
@@ -920,12 +922,24 @@ def _get_children(self, bundle, vol, children, uid):
...
+                    {},
+                ).get('org.freenas:description', ('', '-'))[1]
+
+                # Filtering out comments field for Jails
+                for jail in jails_list:
+                    child_name = child.name.split('/')
+                    length = len(child_name)
+                    if jail.jail_host == child_name[length - 1] or "warden" in child_name[length - 1]:
+                        data['comments'] = '-'
+
```
Comment (williambr) [inline gui/api/resources.py]: No, this doesn't make sense.

You should be looking at the source attribute of the property looking for inherited ones and discard it for org.freenas:description

.e.g.
# zfs get -o source 'org.freenas:description zmedia/jails
SOURCE
inherited from zmedia

If source starts with "inherited" for that property, do not show it.

**Comment by neharathod** [comment]
Okay, I understand that I need to get the source of the jails. Do you want me to create a separate zfs_get_source() in middleware/notifier.py or I can add the piece of code in the same api/resources.py file?

**Comment by neharathod** [comment]
Okay, there exists a method zfs_get_option() that also gets the source, I can set the required validation there

File: gui/api/resources.py [hunk 2]
```diff
@@ -920,12 +922,24 @@ def _get_children(self, bundle, vol, children, uid):
...
                     child.path,
                     {},
                 ).get('readonly', ('', '-'))[1]
 
+                data['comments'] = self.__zfsopts.get(
+                    child.path,
+                    {},
+                ).get('org.freenas:description', ('', '-'))[1]
+
+                # Filtering out comments field for Jails
```
Comment (william-gr) [inline gui/api/resources.py]: Why are you filtering by jail? That doesn't make sense. Its a feature self-contained.

**Pull request closed without merging**